src/ts/entitysystem/components/drawing/ImageDrawable.ts

```diff
@@ -0,0 +1,168 @@
+///<reference path="./Drawable.ts"/>
+///<reference path="../../../util/JsonLoader.ts"/>
+module entityframework.components.drawing {
+
+    import observe = framework.observe;
+    import serialize = util.serialize;
+
+    /**
+     * Represents a drawable that can draw an image.
+     */
+    @serialize.ProvideClass(ImageDrawable, "ild::ImageDrawable")
+    export class ImageDrawable extends Drawable {
+        @observe.Primitive(Boolean)
+        @serialize.Ignore
+        private _loaded : boolean = false;
+
+        @observe.Object()
+        private textureRect : math.Rectangle = new math.Rectangle(0, 0, -1, -1);
+
+        @observe.Primitive(String)
+        textureKey : string = "";
+
+        @observe.Primitive(Boolean)
+        isWholeImage : boolean = true;
+
+        constructor(key : string = "") {
+            super(key);
+        }
+
+        protected generateCanvasDisplayObject(resourceManager : util.resource.ResourceManager) : createjs.DisplayObject {
+            var bitmap = null;
+            if (this.loaded) {
+                bitmap = new createjs.Bitmap(resourceManager.getResource(new map.PNGAsset(this.textureKey)));
+
+                var dimensions = new math.Vector(0, 0);
+                if (!this.isWholeImage) {
+                    dimensions.x = this.textureRect.width;
+                    dimensions.y = this.textureRect.height;
+                    bitmap.sourceRect = new createjs.Rectangle(
+                        this.textureRect.left, this.textureRect.top,
+                        this.textureRect.width, this.textureRect.height);
+                } else {
+                    dimensions.x = bitmap.image.width;
+                    dimensions.y = bitmap.image.height;
+                }
+                bitmap.regX = dimensions.x / 2;
+                bitmap.regY = dimensions.y / 2;
+            } else {
+                this.loadImage("", resourceManager);
+            }
+            return bitmap;
+        }
+
+        loadImage(imageFile : string, resourceManager : util.resource.ResourceManager) {
+            var asset = new map.PNGAsset(this.textureKey);
+            if (resourceManager.hasAsset(asset)) {
+                this.onImageLoaded(asset, new math.Vector(
+                    resourceManager.getResource(asset).width,
+                    resourceManager.getResource(asset).height));
+            }
+
+            if (imageFile) {
+                var image = <HTMLImageElement>asset.createDOMElement(imageFile);
+                image.onload = () => {
+                    resourceManager.addAsset(asset, image);
+                    this.onImageLoaded(asset, new math.Vector(image.width, image.height));
+                }
+            }
+        }
+
+        onImageLoaded(asset : map.PNGAsset, dimensions : math.Vector) {
```

Comment:
The onImageLoaded should exist on a command or on the ViewModel.  It feels really weird to have it on the model itself.